File: 30-Days-of-Code/Day-6-Lets-Review/FSharp/solution.fs
open System

let stringSplitter line =
    line
    |> Seq.toArray
    |> Array.mapi (fun i c -> i, c)
    |> Array.partition (fun (i, _) -> i % 2 = 0)
    |> fun (firstHalf, secondHalf) ->
        firstHalf |> Array.map snd |> System.String,
        secondHalf |> Array.map snd |> System.String

Seq.init (Console.ReadLine() |> int) (fun _ -> Console.ReadLine())
|> Seq.map stringSplitter
|> Seq.iter (fun (firstHalf, secondHalf) -> printfn "%s %s" firstHalf secondHalf)

File: 30-Days-of-Code/Day-6-Lets-Review/Swift/solution.swift
var s = readLine()!
var n : Int! = Int(s)

for i in 0...n-1
{
    s = readLine()!
    var arr = Array(s.characters)

    var firstHalf : String = ""
    var secondHalf : String = ""
    
    for j in 0...arr.count-1
    {
        if(j % 2 == 0)
        {
            firstHalf = firstHalf + String(arr[j])
        } 
        else
        {
            secondHalf = secondHalf + String(arr[j])
        }
    }
    
    print("\(firstHalf) \(secondHalf)")
}
